mypy/plugins/dataclasses.py

```diff
@@ -333,6 +333,38 @@ def collect_attributes(self) -> Optional[List[DataclassAttribute]]:
 
             node = sym.node
             assert not isinstance(node, PlaceholderNode)
+
+            if isinstance(node, TypeAlias):
+                ctx.api.fail(
+                    (
+                        'Type aliases inside dataclass definitions '
+                        'are not supported at runtime.'
+                    ),
+                    Context(line=node.line, column=node.column)
+                )
+                # Now do our best to simulate the runtime,
+                # which treates a TypeAlias definition in a dataclass class
+                # as an instance field with a default value.
+                #
+                # Replace the `TypeAlias` node with a `Var` node, so that we do the same.
+                target, fullname = node.target, node.fullname
+                proper_target = get_proper_type(target)
+                var_type: Union[TypeType, AnyType]
+                if isinstance(proper_target, Instance):
+                    var_type = TypeType(proper_target, line=node.line, column=node.column)
+                # Something else -- fallback to Any
+                else:
+                    var_type = AnyType(TypeOfAny.from_error)
+                var = Var(name=fullname, type=var_type)
```

Comment:
The name argument should be a short name (no dots). I think that you can use `node.name` instead of `fullname`.